### PR TITLE
APP-2854 Azure cloud 

### PR DESCRIPTION
**Pull request by pec1985**

### CODE REVIEW

File: cmd/cmdintegration/oauth.go [hunk 1]
```diff
@@ -25,7 +33,7 @@ func (s *Command) OAuthNewAccessToken(integrationName string) (accessToken strin
 	}
 	refresh := s.OAuthRefreshTokens[integrationName]
 	if refresh == "" {
-		return "", fmt.Errorf("requested oauth access token for integration %v, but we don't have refresh token for it", integrationName)
+		return "", fmt.Errorf("requested oauth access token for integration %v, but we don't have refresh token for it %s", integrationName, pjson.Stringify(s.OAuthRefreshTokens))
```
Comment (pec1985) [inline cmd/cmdintegration/oauth.go]: delete json stringify

File: cmd/cmdintegration/oauth.go [hunk 2]
```diff
@@ -47,9 +57,53 @@ func (s *Command) OAuthNewAccessToken(integrationName string) (accessToken strin
...
 
 	_, err = reqs.JSON(req, &res)
+
+	if err == nil && res.Message != "" {
+		err = errors.New(res.Message)
+	}
+
 	if err != nil {
-		return "", fmt.Errorf("could not get new oauth token from pinpoint backend, err %v", err)
+		return "", fmt.Errorf("could not get new oauth token from pinpoint backend, err %v res %s", err, pjson.Stringify(res))
```
Comment (pec1985) [inline cmd/cmdintegration/oauth.go]: delete json stringify

File: cmd/cmdintegration/oauth.go [hunk 3]
```diff
@@ -47,9 +57,53 @@ func (s *Command) OAuthNewAccessToken(integrationName string) (accessToken strin
...
+	}
+	if res.AccessToken != "" && res.RefreshToken != "" {
+		s.OAuthRefreshTokens[integrationName] = res.RefreshToken
+		s.updateTokens(integrationName, res.AccessToken, res.RefreshToken)
```
Comment (pec1985) [inline cmd/cmdintegration/oauth.go]: handle error

File: cmd/cmdservicerunnorestarts/integration_events.go [hunk 4]
```diff
@@ -37,18 +37,19 @@ func (s *runner) handleIntegrationEvents(ctx context.Context) (closefunc, error)
...
 		req := instance.Object().(*agent.IntegrationRequest)
+
 		headers, err := parseHeader(instance.Message().Headers)
 		if err != nil {
 			return nil, fmt.Errorf("error parsing header. err %v", err)
 		}
 		integration := req.Integration
+		// set this for azure
+		if integration.Authorization.Organization == nil {
+			integration.Authorization.Organization = integration.Organization
```
Comment (pec1985) [inline cmd/cmdservicerunnorestarts/integration_events.go]: In my opinion, the organization should be inside the authorization and not in the main integration object.

**Comment by xzandrew** [comment]
Do we want to wait for "Azure needs to be one integration for both sourcecode and work" before we merge this?

**Comment by pec1985** [comment]
About azure, it's a relatively small change, we should merge these changes since the pipeline teams needs them soon

**Comment by pec1985** [comment]
Yes, let's wait for the one integration before merging. Let's keep the PR open until then

**Comment by pec1985** [comment]
TODO: Come back to it when we need it.

**Pull request closed without merging**